Description: Adapt tests to changes in NumPy 1.10 Default casting rule
Origin: Upstream (commit 9b91b1789c8dc81e84c0a8691febbd1e242a81d1)

---
 pint/testsuite/helpers.py     | 8 +++++++-
 pint/testsuite/test_issues.py | 2 +-
 pint/testsuite/test_numpy.py  | 1 +
 3 files changed, 9 insertions(+), 2 deletions(-)

--- a/pint/testsuite/helpers.py
+++ b/pint/testsuite/helpers.py
@@ -2,11 +2,17 @@
 
 from __future__ import division, unicode_literals, print_function, absolute_import
 
+from distutils.version import StrictVersion
+
 from pint.compat import unittest, HAS_NUMPY, HAS_UNCERTAINTIES, NUMPY_VER, PYTHON3
 
 
 def requires_numpy18():
-    return unittest.skipUnless(NUMPY_VER >= '1.8', 'Requires NumPy >= 1.8')
+    return unittest.skipUnless(StrictVersion(NUMPY_VER) >= StrictVersion('1.8'), 'Requires NumPy >= 1.8')
+
+
+def requires_numpy_previous_than(version):
+    return unittest.skipUnless(StrictVersion(NUMPY_VER) < StrictVersion(version), 'Requires NumPy < %s' % version)
 
 
 def requires_numpy():
--- a/pint/testsuite/test_issues.py
+++ b/pint/testsuite/test_issues.py
@@ -404,7 +404,7 @@
         self.assertQuantityAlmostEqual(x + y, 5.1 * ureg.meter)
         self.assertQuantityAlmostEqual(z, 5.1 * ureg.meter)
 
-
+    @helpers.requires_numpy_previous_than('1.10')
     def test_issue94(self):
         ureg = UnitRegistry()
         v1 = np.array([5, 5]) * ureg.meter
--- a/pint/testsuite/test_numpy.py
+++ b/pint/testsuite/test_numpy.py
@@ -165,6 +165,7 @@
         self.assertRaises(ValueError, self.q.cumprod)
         self.assertQuantityEqual((self.q / self.ureg.m).cumprod(), [1, 2, 6, 24])
 
+    @helpers.requires_numpy_previous_than('1.10')
     def test_integer_div(self):
         a = [1] * self.ureg.m
         b = [2] * self.ureg.m
